File: Go/Easy/Generate Document/solution_1.go
package main

func GenerateDocument(characters string, document string) bool {

	for _, character := range document {

		documentFrequency := countFrequency(character, document)
		characterFrequency := countFrequency(character, characters)

		if documentFrequency > characterFrequency {
			return false
		}
	}

	return true

}

func countFrequency(targetCharacter rune, array string) int {

	var frequency = 0
	for _, char := range array {
		if char == targetCharacter {
			frequency++
		}
	}
	return frequency
}

File: Python/Easy/Generate Document/solution_1.py
def generateDocument(characters, document):

    for character in document:

        documentFrequency = countFrequency(character, document)
        characterFrequency = countFrequency(character, characters)

        if documentFrequency > characterFrequency:
            return False
            
    return True


def countFrequency(targetCharacter, array):
    frequency = 0
    for letter in array:
        if letter == targetCharacter:
            frequency += 1
    
    return frequency
